solr/solrj/src/java/org/apache/solr/common/util/JavaBinCodec.java

```diff
@@ -100,6 +100,7 @@
           MAP_ENTRY_ITER = 17,
           ENUM_FIELD_VALUE = 18,
           MAP_ENTRY = 19,
+          UUID = 20,
```

Comment:
We should not add new types unless absolutely required. I'm not sure `UUID` is requires a new type

Reply:
As you can see in the provided stack trace, it is required that JavaBinCodec must handle UUID, because it is written to the tlog. If you have a suggestion how to marshal and unmarshal such a type without using an type identifier, plz tell me, i did not see such a possibility

Reply:
why does the unmarshal need to know the type?

Reply:
because it needs to the exact opposite operation of the marshalling, otherwise the object can not be constructed from the stream

Reply:
If I need to index a `UUID` field , I send a `String` using JSON and it works just fine right? How does it know how to handle the `UUID` field? The point is, there will be so many different types of Objects we need to handle.  An example is Point/Shape fields. javabin may not be able to accommodate all of them. It makes the format more complicated

Reply:
if you do this, you will index it as StringField not as UUIDField

Reply:
It depends on the type of the field , right?

If I send a `String` to `myfield_uuid ` which is configured as a UUID type, Solr will internally convert it to UUID , right?

Reply:
Now i see your point, you are talking about the usage of JavaBinCodec used by the request handler, this is different to the usage of transaction logs (where the JavaBinCodec is also used), which requires marshalling and unmarshalling of the UUID type

Reply:
Starting point is TransactionLog.write(TransactionLog.java:371)